### DISCUSSION TITLE
CREATE OR REPLACE TRIGGER does not validate in SQL editor

### DISCUSSION THREAD

**Discussion by tanius** [supabase/supabase#16926] [Contribute to Supabase]
# Bug report

## Describe the bug

`CREATE OR REPLACE TRIGGER ...` is valid syntax since PostgreSQL 14 ([see](https://www.postgresql.org/docs/14/sql-createtrigger.html)). Such queries work in Supabase databases when executed via `psql`. However, such a query is not accepted by the SQL editor on app.supabase.com:

```none
Failed to validate sql query: syntax error at or near "TRIGGER"
```

## To Reproduce

1. Go to any of your projects on app.supabase.com
2. Click on "SQL Editor" in the left column.
3. Enter and execute a valid `CREATE OR REPLACE TRIGGER` query, such as:

    ```sql
    CREATE OR REPLACE TRIGGER insert_profile AFTER INSERT ON auth.users
    FOR EACH ROW EXECUTE FUNCTION public.insert_profile();
    ```

## Expected behavior

The Supabase SQL Editor should accept the query, since it's valid SQL. It probably needs to be updated. Similar problems may exist with other syntax that is valid since PostgreSQL 14 (or even earlier).

**Comment by saltcod** [comment]
Limited by the parser itself:
https://github.com/pyramation/pgsql-parser/issues/93
